### ISSUE TITLE
GUI Crashed Resuting in GLib-GIO-ERROR 

### ISSUE THREAD

**Issue by brewersey** [conda-forge/gnuradio-feedstock#24]
Issue:
When I click on a menu option in gnuradio-companion, such as <File --> Open> the GUI crashes, giving the error:
(gnuradio-companion:4292): GLib-GIO-ERROR **: 09:57:12.172: Settings schema 'org.gtk.Settings.FileChooser' does not contain a key named 'type-format'
Trace/breakpoint trap (core dumped)

My operating system is Ubuntu:
Distributor ID:	Ubuntu
Description:	Ubuntu 18.04.4 LTS
Release:	18.04
Codename:	bionic

Thanks!
<br/>
Environment (<code>conda list</code>):
<details>
...
$ conda list
#
# Name                    Version                   Build  Channel
_libgcc_mutex             0.1                        main  
adwaita-icon-theme        3.36.1               h5728ca3_0    conda-forge
alsa-lib                  1.1.5             h516909a_1002    conda-forge
at-spi2-atk               2.32.0               h1369247_0    conda-forge
at-spi2-core              2.35.1               h1369247_0    conda-forge
atk                       2.36.0                        0    conda-forge
atk-1.0                   2.36.0               haf93ef1_0    conda-forge
attrs                     19.3.0                     py_0  
backcall                  0.2.0                      py_0  
blas                      1.0                         mkl  
bleach                    3.1.5                      py_0  
bokeh                     2.1.0                    py37_0  
boost-cpp                 1.72.0               h8e57a91_0    conda-forge
brotlipy                  0.7.0           py37h7b6447c_1000  
bzip2                     1.0.8                h7b6447c_0  
ca-certificates           2020.1.1                      0  
cairo                     1.16.0            hcf35c78_1003    conda-forge
certifi                   2020.4.5.2               py37_0  
cffi                      1.14.0           py37h2e261b9_0  
chardet                   3.0.4                 py37_1003  
click                     7.1.2                      py_0  
click-plugins             1.1.1                      py_0  
cloudpickle               1.4.1                      py_0  
cryptography              2.9.2            py37h1ba5d50_0  
cytoolz                   0.10.1           py37h7b6447c_0  
dask                      2.18.1                     py_0  
dask-core                 2.18.1                     py_0  
dbus                      1.13.14              hb2f20db_0  
decorator                 4.4.2                      py_0  
defusedxml                0.6.0                      py_0  
distributed               2.18.0                   py37_0  
entrypoints               0.3                      py37_0  
epoxy                     1.5.4                h516909a_1    conda-forge
expat                     2.2.6                he6710b0_0  
fftw                      3.3.8                h7b6447c_3  
fontconfig                2.13.1            h86ecdb6_1001    conda-forge
freetype                  2.10.2               h5ab3b9f_0  
fribidi                   1.0.9                h516909a_0    conda-forge
fsspec                    0.7.4                      py_0  
gdk-pixbuf                2.38.2               h3f25603_3    conda-forge
gettext                   0.19.8.1             hd7bead4_3  
glib                      2.63.1               h5a9c865_0  
gmp                       6.1.2                h6c8ec71_1  
gnuradio                  3.8.1.0          py37h4c8c553_6    conda-forge
gnuradio-core             3.8.1.0          py37h615a080_6    conda-forge
gnuradio-grc              3.8.1.0          py37h110162e_6    conda-forge
gnuradio-osmosdr          0.2.0            py37h110162e_1    ryanvolz
gnuradio-qtgui            3.8.1.0          py37he471202_6    conda-forge
gnuradio-uhd              3.8.1.0          py37h110162e_6    conda-forge
gnuradio-video-sdl        3.8.1.0          py37h110162e_6    conda-forge
gnuradio-zeromq           3.8.1.0          py37h913d78c_6    conda-forge
gobject-introspection     1.56.1           py37hbc4ca2d_2  
graphite2                 1.3.13               h23475e2_0  
gsl                       2.6                  h294904e_0    conda-forge
gst-plugins-base          1.14.5               h0935bb2_2    conda-forge
gstreamer                 1.14.5               h36ae1b5_2    conda-forge
gtk3                      3.24.14              h45fd312_2    conda-forge
harfbuzz                  2.4.0                h9f30f68_3    conda-forge
heapdict                  1.0.1                      py_0  
hicolor-icon-theme        0.17                          0    conda-forge
icu                       64.2                 he1b5a44_1    conda-forge
idna                      2.9                        py_1  
importlib-metadata        1.6.1                    py37_0  
importlib_metadata        1.6.1                         0  
intel-openmp              2019.4                      243  
ipykernel                 5.3.0            py37h5ca1d4c_0  
ipython                   7.15.0                   py37_0  
ipython_genutils          0.2.0                    py37_0  
jedi                      0.17.0                   py37_0  
jinja2                    2.11.2                     py_0  
jpeg                      9c                h14c3975_1001    conda-forge
jsonschema                3.2.0                    py37_0  
jupyter_client            6.1.3                      py_0  
jupyter_core              4.6.3                    py37_0  
krb5                      1.16.4               h173b8e3_0  
ld_impl_linux-64          2.33.1               h53a641e_7  
libblas                   3.8.0                    14_mkl    conda-forge
libcblas                  3.8.0                    14_mkl    conda-forge
libclang                  9.0.1           default_hde54327_0    conda-forge
libcroco                  0.6.13               h8d621e5_1    conda-forge
libcups                   2.2.12               h6ffcb39_0    conda-forge
libedit                   3.1.20181209         hc058e9b_0  
libffi                    3.2.1                hd88cf55_4  
libgcc-ng                 9.1.0                hdf63c60_0  
libgfortran-ng            7.3.0                hdf63c60_0  
libiconv                  1.15                 h63c8f33_5  
libllvm9                  9.0.1                he513fc3_1    conda-forge
libpng                    1.6.37               hbc83047_0  
librsvg                   2.48.4               h33a7fed_0    conda-forge
libsodium                 1.0.17               h516909a_0    conda-forge
libstdcxx-ng              9.1.0                hdf63c60_0  
libtiff                   4.1.0                h2733197_1  
libusb                    1.0.23               he1b5a44_1    conda-forge
libuuid                   2.32.1            h14c3975_1000    conda-forge
libxcb                    1.13                 h1bed415_1  
libxkbcommon              0.10.0               he1b5a44_0    conda-forge
libxml2                   2.9.10               hee79883_0    conda-forge
libxslt                   1.1.33               h31b3aaa_0    conda-forge
locket                    0.2.0                    py37_1  
log4cpp                   1.1.3             hf484d3e_1001    conda-forge
lxml                      4.5.1            py37hefd8a0e_0  
lz4-c                     1.9.2                he6710b0_0  
mako                      1.1.3                      py_0  
markupsafe                1.1.1            py37h7b6447c_0  
mistune                   0.8.4            py37h7b6447c_0  
mkl                       2019.4                      243  
mkl-service               2.3.0            py37he904b0f_0  
mkl_fft                   1.1.0            py37h23d657b_0  
mkl_random                1.1.0            py37hd6b4f25_0  
mpir                      3.0.0                h3e5f119_1  
msgpack-python            1.0.0            py37hfd86e86_1  
nbconvert                 5.6.1                    py37_0  
nbformat                  5.0.7                      py_0  
ncurses                   6.1                  he6710b0_1  
notebook                  6.0.3                    py37_0  
nspr                      4.22                 he6710b0_0  
nss                       3.46.1               hab99668_0  
numpy                     1.18.1           py37h4f9e942_0  
numpy-base                1.18.1           py37hde5b4d6_1  
olefile                   0.46                     py37_0  
openssl                   1.1.1g               h7b6447c_0  
packaging                 20.4                       py_0  
pandas                    1.0.4            py37h0573a6f_0  
pandoc                    2.2.3.2                       0  
pandocfilters             1.4.2                    py37_1  
pango                     1.42.4               h7062337_4    conda-forge
parso                     0.7.0                      py_0  
partd                     1.1.0                      py_0  
pcre                      8.43                 he6710b0_0  
pexpect                   4.8.0                    py37_0  
pickleshare               0.7.5                    py37_0  
pillow                    7.1.2            py37hb39fc2d_0  
pip                       20.1.1                   py37_1  
pixman                    0.38.0               h7b6447c_0  
portaudio                 19.6.0               h7b6447c_4  
prometheus_client         0.8.0                      py_0  
prompt-toolkit            3.0.5                      py_0  
psutil                    5.7.0            py37h7b6447c_0  
ptyprocess                0.6.0                    py37_0  
pycairo                   1.19.0           py37h2a1e443_0  
pycparser                 2.20                       py_0  
pygments                  2.6.1                      py_0  
pygobject                 3.30.5           py37hfc86313_1    conda-forge
pyopenssl                 19.1.0                   py37_0  
pyparsing                 2.4.7                      py_0  
pyqt                      5.12.3           py37h8685d9f_3    conda-forge
pyqt5-sip                 4.19.18                  pypi_0    pypi
pyqtchart                 5.12                     pypi_0    pypi
pyqtwebengine             5.12.1                   pypi_0    pypi
pyrsistent                0.16.0           py37h7b6447c_0  
pysocks                   1.7.1                    py37_0  
python                    3.7.6                h0371630_2  
python-dateutil           2.8.1                      py_0  
python_abi                3.7                     1_cp37m    conda-forge
pytz                      2020.1                     py_0  
pyyaml                    5.3.1            py37h7b6447c_0  
pyzmq                     18.1.1           py37he6710b0_0  
qt                        5.12.5               hd8c4c69_1    conda-forge
qwt                       6.1.4             h73816c6_1005    conda-forge
readline                  7.0                  h7b6447c_5  
requests                  2.23.0                   py37_0  
rtl-sdr                   0.6.0                h516909a_0    conda-forge
sdl                       1.2.15               he1b5a44_0    conda-forge
send2trash                1.5.0                    py37_0  
setuptools                47.3.0                   py37_0  
six                       1.15.0                     py_0  
soapysdr                  0.7.2            py37h99015e2_1    conda-forge
sortedcontainers          2.2.2                      py_0  
sqlite                    3.31.1               h7b6447c_0  
tblib                     1.6.0                      py_0  
terminado                 0.8.3                    py37_0  
testpath                  0.4.4                      py_0  
tk                        8.6.10               hbc83047_0  
toolz                     0.10.0                     py_0  
tornado                   6.0.4            py37h7b6447c_1  
traitlets                 4.3.3                    py37_0  
typing_extensions         3.7.4.1                  py37_0  
uhd                       3.15.0.0         py37h9598972_2    conda-forge
urllib3                   1.25.9                     py_0  
volk                      2.3.0            py37h3340039_1    conda-forge
wcwidth                   0.2.4                      py_0  
webencodings              0.5.1                    py37_1  
wheel                     0.34.2                   py37_0  
xorg-kbproto              1.0.7             h14c3975_1002    conda-forge
xorg-libice               1.0.10               h516909a_0    conda-forge
xorg-libsm                1.2.3             h84519dc_1000    conda-forge
xorg-libx11               1.6.9                h516909a_0    conda-forge
xorg-libxext              1.3.4                h516909a_0    conda-forge
xorg-libxrender           0.9.10            h516909a_1002    conda-forge
xorg-renderproto          0.11.1            h14c3975_1002    conda-forge
xorg-xextproto            7.3.0             h14c3975_1002    conda-forge
xorg-xproto               7.0.31            h14c3975_1007    conda-forge
xz                        5.2.5                h7b6447c_0  
yaml                      0.1.7                had09818_2  
zeromq                    4.3.2                he1b5a44_2    conda-forge
zict                      2.0.0                      py_0  
zipp                      3.1.0                      py_0  
zlib                      1.2.11               h7b6447c_3  
zstd                      1.4.4                h0b5b093_3  

</details>

<br/>
Details about  <code>conda</code> and system ( <code>conda info</code> ):
<details>
...
$ conda info
     active environment : radio
    active env location : /home/normanjrt/anaconda3/envs/radio
            shell level : 2
       user config file : /home/normanjrt/.condarc
 populated config files : /home/normanjrt/.condarc
          conda version : 4.8.3
    conda-build version : 3.18.11
         python version : 3.7.4.final.0
       virtual packages : __glibc=2.27
       base environment : /home/normanjrt/anaconda3  (writable)
           channel URLs : https://repo.anaconda.com/pkgs/main/linux-64
                          https://repo.anaconda.com/pkgs/main/noarch
                          https://repo.anaconda.com/pkgs/r/linux-64
                          https://repo.anaconda.com/pkgs/r/noarch
          package cache : /home/normanjrt/anaconda3/pkgs
                          /home/normanjrt/.conda/pkgs
       envs directories : /home/normanjrt/anaconda3/envs
                          /home/normanjrt/.conda/envs
               platform : linux-64
             user-agent : conda/4.8.3 requests/2.22.0 CPython/3.7.4 Linux/4.15.0-111-generic ubuntu/18.04.4 glibc/2.27
                UID:GID : 1000:1000
             netrc file : None
           offline mode : False

```
</details>

**Comment by ryanvolz** [comment]
Thanks for the report. This is likely to be an issue with `gtk3`, but let's see if we can debug.

Can you inspect the contents of `"${CONDA_PREFIX}/share/glib-2.0/schemas"` and save the output of `ls -lf` on that directory?

Then try running this command and see if GRC will work after that:
```
"${CONDA_PREFIX}/bin/glib-compile-schemas" "${CONDA_PREFIX}/share/glib-2.0/schemas"
```

Finally, inspect that directory again after running the command and see if anything has changed.

That command is something that is supposed to be run automatically after `gtk3` is installed, but based on the error message I'm wondering if it failed.

**Comment by ryanvolz** [comment]
These also seem relevant:

https://askubuntu.com/questions/820733/firefox-hangs-because-sort-directories-first-setting-missing
https://askubuntu.com/questions/164570/broken-schema-org-gtk-settings-filechooser

The conda gtk3 might be picking up files from your system installation that are interfering.

**Comment by brewersey** [comment]
Hi Ryan,
When I executed the "${CONDA_PREFIX}/bin/glib-compile-schemas ${CONDA_PREFIX}/share/glib-2.0/schemas" I did not see a change in the output of ls -lf.

I reviewed the links that you suggested.  Then uninstalled gtk3 then installed the anaconda-forge gtk3:
conda install -c conda-forge gtk3

By uninstalling gtk3 many gnuradio components were also uninstalled, so I had to reinstall gnuradio after I installed the anaconda-forge gtk3.  This fixed the problem, and I can now run grc with out issue.

Thanks for you help and advice!
Thaddeus
